File: server/src/main/java/org/hkurh/doky/entities/UserEntity.java
package org.hkurh.doky.entities;

import javax.persistence.*;

@Entity
@Table(name = "user")
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "uid", nullable = false, unique = true)
    private String uid;

    @Column(name = "name")
    private String name;

    @Column(name = "password", nullable = false)
    private String password;

    public String getPassword() {
        return password;
    }

    public void setPassword(final String password) {
        this.password = password;
    }

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getUid() {
        return uid;
    }

    public void setUid(final String uid) {
        this.uid = uid;
    }

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }
}

File: server/src/main/java/org/hkurh/doky/users/UserEntity.kt
package org.hkurh.doky.users

import jakarta.persistence.*

@Entity
@Table(name = "user",
        indexes = [Index(name = "idx_userentity_uid", columnList = "uid")],
        uniqueConstraints = [UniqueConstraint(name = "uc_userentity_uid", columnNames = ["uid"])])
class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Long = -1

    @Column(name = "uid", nullable = false, unique = true)
    var uid: String = ""

    @Column(name = "name")
    var name: String? = null

    @Column(name = "password", nullable = false)
    var password: String = ""
}
